resources/shared-db/src/mongo.rs

```diff
@@ -1,46 +1,48 @@
 use async_trait::async_trait;
-use serde::Serialize;
+use serde::{Deserialize, Serialize};
 use shuttle_service::{
-    database, error::CustomError, DbInput, DbOutput, Error, Factory, ResourceBuilder, Type,
+    database, error::CustomError, resource::Type, DatabaseResource, DbInput, Error, Factory,
+    IntoResource, ResourceBuilder,
 };
 
-#[derive(Serialize)]
-pub struct MongoDb {
-    config: DbInput,
+/// Shuttle managed MongoDB in a shared cluster
+#[derive(Default)]
+pub struct MongoDb(DbInput);
```

Comment:
Nit: is there a good reason to switch to a tuple struct here, or could we keep it as a struct with a named field, for increased readability?

Reply:
Both `MongoDb` and `Postgres` were basically following the newtype pattern already, with just one named field. This felt more coherent, but doesn't matter at the end of the day.